src/custom Hook/useApi.tsx

```diff
@@ -0,0 +1,43 @@
+import axios from "axios"
+import { useContext, useEffect, useState } from "react"
+import { ToastContext } from "../Context/ToastContext"
+
+const useApi =(url)=>{
```

Comment:
great job on creating a custom hook, but it would be better to separate the concerns. Instead of having a generic `useAPI`, you can create a custom hook that is only responsible for the facilities and call it `useFacilities` and another one for `useRooms` responsible for the rooms. This will make the code easier to read and maintain.